cranelift/codegen/src/write.rs

```diff
@@ -380,7 +382,17 @@ pub fn write_operands(w: &mut dyn Write, dfg: &DataFlowGraph, inst: Inst) -> fmt
         LoadNoOffset { flags, arg, .. } => write!(w, "{} {}", flags, arg),
         StoreNoOffset { flags, args, .. } => write!(w, "{} {}, {}", flags, args[0], args[1]),
         Unary { arg, .. } => write!(w, " {}", arg),
-        UnaryImm { imm, .. } => write!(w, " {}", imm),
+        UnaryImm { imm, .. } => write!(
+            w,
+            " {}",
+            Imm64::new(match dfg.ctrl_typevar(inst) {
+                types::I8 => imm.bits() as i8 as i64,
+                types::I16 => imm.bits() as i16 as i64,
+                types::I32 => imm.bits() as i32 as i64,
+                types::I64 => imm.bits(),
+                _ => unreachable!(),
+            })
+        ),
```

Comment:
The handful of test failures on your branch that are in `filetests/filetests/egraph/` are because of this change: constants which were narrower than 64 bits were previously printed as unsigned, and are now printed as signed, but these tests expect the printed text to match what's in the test.

This change is harmless and I kind of like having these values printed the way you've done it, so you can fix the egraph tests by updating them to match the new format.

Reply:
OK! 🙂 The change in `write.rs` was needed to fix a test case, but ended up breaking some others, as you said. I'll update those outputs.